src/tallies/tally_filter_zernike_radial.F90

```diff
@@ -0,0 +1,203 @@
+module tally_filter_zernike_radial
+
+  use, intrinsic :: ISO_C_BINDING
+
+  use constants
+  use error
+  use hdf5_interface
+  use math,                only: calc_zn_rad
+  use particle_header,     only: Particle
+  use string,              only: to_str
+  use tally_filter_header
+  use xml_interface
+
+  implicit none
+  private
+
+!===============================================================================
+! ZERNIKERADIALFILTER gives even order radial Zernike polynomial moments of a 
+! particle's position
+!===============================================================================
+
+  type, public, extends(TallyFilter) :: ZernikeRadialFilter
```

Comment:
As with the Python code, I think `ZernikeRadialFilter` could be defined to `extends(ZernikeFilter)` which would remove the need to define the parameters order, x, y, and r, as well as the `from_xml` procedure (you can look at `EnergyFilter` and `EnergyoutFilter` for a similar example).

Reply:
Did you mean Fortran code?